kedro-datasets/kedro_datasets_experimental/safetensors/safetensors_dataset.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import annotations
+
+import importlib
+from copy import deepcopy
+from pathlib import PurePosixPath
+from typing import Any
+
+import fsspec
+from kedro.io.core import (
+    AbstractVersionedDataset,
+    DatasetError,
+    Version,
+    get_filepath_str,
+    get_protocol_and_path,
+)
+
+
+class SafetensorsDataset(AbstractVersionedDataset[Any, Any]):
+    """``SafetensorsDataset`` loads/saves data from/to a Safetensors file using an underlying
+    filesystem (e.g.: local, S3, GCS). The underlying functionality is supported by
+    the specified backend library passed in (defaults to the ``torch`` library), so it
+    supports all allowed options for loading and Safetensors files.
+
+    Example usage for the
+    `YAML API <https://docs.kedro.org/en/stable/data/\
+    data_catalog_yaml_examples.html>`_:
+
+    .. code-block:: yaml
+
+        test_model: # simple example without compression
+          type: safetensors.SafetensorsDataset
+          filepath: data/07_model_output/test_model.safetensors
+          backend: torch
+
+    Example usage for the
+    `Python API <https://docs.kedro.org/en/stable/data/\
+    advanced_data_catalog_usage.html>`_:
+
+    .. code-block:: pycon
+
+        >>> from kedro_datasets.safetensors import SafetensorsDataset
+        >>> import torch
+        >>>
+        >>> data = {"embeddings": torch.zeros((10, 100)}
+        >>> dataset = SafetensorsDataset(
+        ...     filepath="test.safetensors",
+        ...     backend="torch"
+        ... )
+        >>> dataset.save(data)
+        >>> reloaded = dataset.load()
+        >>> assert data.equals(reloaded)
```

Comment:
```suggestion
        >>> from kedro_datasets_experimental.safetensors import SafetensorsDataset
        >>> import torch
        >>>
        >>> data = {"embeddings": torch.zeros((10, 100)}
        >>> dataset = SafetensorsDataset(
        ...     filepath="test.safetensors",
        ...     backend="torch"
        ... )
        >>> dataset.save(data)
        >>> reloaded = dataset.load()
        >>> assert data.equals(reloaded)
```

`assert data.equals(reloaded)` this gives me an error:
> AttributeError: 'dict' object has no attribute 'equals'

Reply:
Hey @noklam,
I made some updates to the example in the docstring. The above error should be fixed now.